torch/csrc/jit/register_special_ops.cpp

```diff
@@ -283,6 +347,29 @@ RegisterOperators reg({
           push(stack, std::move(tensor));                                     \
           return 0;                                                           \
         };                                                                    \
+      }),                                                                     \
+  Operator(                                                                   \
+      "aten::as_tensor(" #operator_type                                       \
+      " t, *, ScalarType? dtype=None, Device? device=None) -> Tensor",        \
```

Comment:
Can you refactor this function with the equivalent torch::tensor one ?

Reply:
I was trying to do that, then I realized that there're macro's arg that could not be placed just as a template arg, like `tensor_creation_op`

Reply:
let me try it again to see if it works.

Reply:
Again failed because `tensor_creation_op` is a function call instance that could not make into a template arg.

Reply:
Yea makes sense. Could you factor out 
```
    at::ScalarType scalar_type =
        dtype.isNone() ? tensor.scalar_type() : dtype.toScalarType();
    c10::Device dev =
        device.isNone() ? tensor.device() : device.toDevice();
    if (scalar_type != initial_scalar_type || dev != tensor.device()) {
      tensor = tensor.to(dev, scalar_type);
    }
``` 
to a method ? those lines are repeated 3 times in the file.